File: data/transcoder_evaluation_gfg/go/MAXIMIZE_VOLUME_CUBOID_GIVEN_SUM_SIDES.go
package main

import (
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(s int) int {
	var maxvalue int = 0
	for i := int(1); i <= s-2; i++ {
		for j := int(1); j <= s-1; j++ {
			var k int = s - i - j
			maxvalue = max(maxvalue, i*j*k)
		}
	}
	return maxvalue
}
//TOFILL
func main() {
	var (
		n_success int     = 0
		param0    []int = []int{67, 48, 59, 22, 14, 66, 1, 75, 58, 78}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if f_filled(param0[i]) == f_gold(param0[i]) {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/rust/MAXIMIZE_VOLUME_CUBOID_GIVEN_SUM_SIDES.rs
#![feature(main)]
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//

#[no_mangle]
pub unsafe fn abs(mut x: i32)
 -> i32 {
    return x.abs();
}

#[no_mangle]
pub unsafe fn min(mut x: i32, mut y: i32)
 -> i32 {
    return if x < y { x } else { y };
}
#[no_mangle]
pub unsafe fn max(mut x: i32, mut y: i32)
 -> i32 {
    return if x > y { x } else { y };
}

pub unsafe fn f_gold(mut s: i32) -> i32 {
    let mut maxvalue: i32 = 0 as i32;
    let mut i: i32 = 1 as i32;
    while i <= s - 2 as i32 {
        let mut j: i32 = 1 as i32;
        while j <= s - 1 as i32 {
            let mut k: i32 = s - i - j;
            maxvalue = max(maxvalue, i * j * k);
            j += 1
        }
        i += 1
    }
    return maxvalue;
}
//TOFILL
unsafe fn main_0() -> i32 {
    let mut n_success: i32 = 0 as i32;
    let mut param0: [i32; 10] =
        [67 as i32, 48 as i32, 59 as i32,
         22 as i32, 14 as i32, 66 as i32,
         1 as i32, 75 as i32, 58 as i32,
         78 as i32];
    let mut i: i32 = 0 as i32;
    while i < param0.len() as i32 {
        if f_filled(param0[i as usize]) == f_gold(param0[i as usize]) {
            n_success += 1 as i32
        }
        i += 1
    }
    println!("{} {} {} {} {}", "#Results:", " ", n_success, ", ", param0.len() as i32);
    return 0 as i32;
}
#[main]
pub fn main() { unsafe { ::std::process::exit(main_0() as i32) } }
